### PR TITLE
chore: update Django constraint to match latest LTS version

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
**Description:**

This PR updates the Django common constraint used by other projects' repositories to use the latest LTS version: https://openedx.atlassian.net/wiki/spaces/AC/pages/3767926815/Django+4.2+Upgrade+Plan

**Merge checklist:**
- [ ] All reviewers approved
- [x] CI build is green
- [x] If adding new checks, followed how_tos/0001-adding-new-checks.rst
- [ ] Changelog record added (if needed) NA
- [ ] Documentation updated (not only docstrings) (if needed) NA
- [x] Commits are squashed

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @mariajgrimaldi! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by mphilbrick211** [comment]
@feanil - is this something Axim should review?

**Comment by feanil** [comment]
Yea, there are still 2 services that are not running django 4 so we'll need to update the local constraints for those services so we can merge this, I'll reach out to the team that's doing the upgrades for those final 2 services.

File: edx_lint/files/common_constraints.txt [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 
 # using LTS django version
-Django<4.0
+Django<5.0
```
[the remark this answers is not in the document]
Comment (UsamaSadiq) [reply edx_lint/files/common_constraints.txt:16]: We are tracking the pending repos in the issue https://github.com/edx/edx-arch-experiments/issues/552 and are currently creating PRs in those repos.

**Review by UsamaSadiq — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@mariajgrimaldi 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-lint&entry.867055334=https://github.com/openedx/edx-lint/pull/403&entry.1484655318=https://github.com/mariajgrimaldi&entry.752974735=2024-02-27+12:56&entry.1917517419=2024-02-29+13:44&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**